exporter/awss3exporter/s3_writer.go

```diff
@@ -39,12 +49,14 @@ func newUploadManager(
 				DisableHTTPS: conf.S3Uploader.DisableSSL,
 			}
 			o.UsePathStyle = conf.S3Uploader.S3ForcePathStyle
+			o.Retryer = retry.AddWithMaxAttempts(o.Retryer, conf.S3Uploader.RetryMaxAttempts)
```

Comment:
The SDK provides too many ways to config the same thing, but it seems this util wrapper func is the only way to set the max attempts to 0 and allow unlimited retries. Init the retrier with `NewStandard` or overwrite `StandardOptions` actually doesn't allow setting `MaxAttempts` to 0: https://github.com/aws/aws-sdk-go-v2/blob/c1c1cda2978dcef4de6db7747e2f427105023cde/aws/retry/standard.go#L186-L188.

I even doubted if the documentation was lying about it, but it turns out setting `MaxAttempts` to 0 indeed allows unlimited retries: https://github.com/aws/aws-sdk-go-v2/blob/c1c1cda2978dcef4de6db7747e2f427105023cde/aws/retry/middleware.go#L248-L255

I find it quite confusing and would like a fresh pair of eyes to double check.

Reply:
This matches my expectations. To be fair, this would work with a negative value as well, which might seem less confusing.